exec/stages/prepivot/prepivot.go

```diff
@@ -58,6 +58,22 @@ func (s stage) Run(config config.Config) bool {
 		if !s.writeUnit(unit) {
 			return false
 		}
+		if unit.Enable {
+			s.logger.Info(fmt.Sprintf("enabling unit %q", unit.Name))
+			if err := util.EnableUnit(s.root, unit); err != nil {
+				s.logger.Info(fmt.Sprintf("failed to enable unit %q: %v", unit.Name, err))
+				return false
+			}
+			s.logger.Info(fmt.Sprintf("done enabling unit %q", unit.Name))
+		}
+		if unit.Mask {
+			s.logger.Info(fmt.Sprintf("masking unit %q", unit.Name))
+			if err := util.MaskUnit(s.root, unit); err != nil {
+				s.logger.Info(fmt.Sprintf("failed to mask unit %q: %v", unit.Name, err))
+				return false
+			}
+			s.logger.Info(fmt.Sprintf("done masking unit %q", unit.Name))
+		}
```

Comment:
this is such copy-and-paste code it's annoying, the two clauses are pretty much identical except for what they're conditional upon, the log messages, and what function is called conditionally.

In C I'd have a macro for this crap, can we do something else in Golang to avoid the repetition without much contortion?